File: problems/Easy/python/twoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ComplementMap = dict()
        for i, num  in enumerate(nums):
            complement = target - num

            if num in ComplementMap:
                return [ComplementMap[num], i]
            else:
                ComplementMap[complement] = i

File: problems/Easy/javascript/twosum.js
//* Initial Solve n^2 prolly

var twoSum = function (nums, target) {
	for (let i = 0; i < nums.length; i++) {
		for (let j = nums.length; j >= i + 1; j--) {
			if (nums[i] + nums[j] === target) {
				return [i, j]

			}
		}
	}

};

//* Using complements

var twoSum = function (nums, target) {
	for (let i = 0; i < nums.length; i++) {
		let val = target - nums[i];
		let j = nums.indexOf(val);

		if (j !== -1 && j !== i) {
			return [i, j]
		}
	}

};

//# JS Is slower than python 